structured-event-log/src/main/java/org/apache/pulsar/structuredeventlog/slf4j/Slf4jEvent.java

```diff
@@ -160,7 +166,15 @@ public void log(String event) {
                 MDC.put("startTimestamp", startTime.toString());
                 MDC.put("durationMs", String.valueOf(Duration.between(startTime, clock.instant()).toMillis()));
             }
-            Logger logger = LoggerFactory.getLogger("stevlog");
+            Logger logger;
+            if (component != null) {
+                MDC.put("component", component);
+                logger = LoggerFactory.getLogger(
+                        new StringBuilder("stevlog.").append(component)
+                        .append(".").append(event).toString());
```

Comment:
If this part of the code is a hotspot, it would be better if the logging code is mostly "garbage free". Would it be possible to refactor this so that the logger name isn't created for each call and looked up?

Reply:
will change it to use a hashmap in TLS

Reply:
@lhotari new change up

Reply:
Elegant solution!